### PR TITLE
Don't delete stacks indiscriminately

### PR DESCRIPTION
**Pull request by arbrandes**
Assume a rollback stance when stack launching fails: if the failure
occurred during creation, delete it; during resuming, suspend it.

For clarity, also separate the stack launch status from the verification
status as much as possible.

TG-3 #closed

### CODE REVIEW

File: hastexo/tasks.py [hunk 1]
```diff
@@ -99,7 +107,8 @@ def launch_stack(self, heat, stack_name, stack_template):
             status = stack.stack_status
             retry += 1
             if retry >= self.retries:
-                status = 'CREATE_FAILED'
+                status_prefix = status[:-12]
```
Comment (fghaas) [inline hastexo/tasks.py]: Not sure where the `-12` comes from. Also, would probably be nicer (and more maintainable) to use a regex match/replace here.

**Comment by fghaas** [comment]
It would be really helpful if the failure messages would include the stack UUID on top of the stack name. Once a stack is deleted, it's non-trivial to find out what its UUID was, but only the UUID would show up in the cloud provider's logs.

File: hastexo/tasks.py (hunk 1 above)
Comment (arbrandes) [inline hastexo/tasks.py]: -12 is to remove the suffix from *_IN_PROGRESS, the only possible statuses here.  But ok, a regex sub will do just as well.

**Comment by arbrandes** [comment]
While the last comment makes perfect sense, this warrants a separate user story (or extending TG-1).  It's not always trivial to get the stack UUID from within the XBlock, either.

File: hastexo/tasks.py (hunk 1 above)
Comment (arbrandes) [inline hastexo/tasks.py]: Fixed on latest PR.

**Comment by fghaas** [comment]
Sure, putting stack UUID logging into its own US is fine.

**Pull request merged**